src/Autofac/Core/Activators/Reflection/ReflectionActivator.cs

```diff
@@ -91,6 +93,20 @@ public void ConfigurePipeline(IComponentRegistryServices componentRegistryServic
             binders[idx] = new ConstructorBinder(availableConstructors[idx]);
         }
 
+        if (binders.Length == 1)
+        {
+            UseSingleConstructorActivation(pipelineBuilder, binders[0]);
+            return;
+        }
+        else if (ConstructorSelector is MatchingSignatureConstructorSelector matchingSelector)
```

Comment:
It feels kind of... weird? bad?... to me that we have to "know" about a specific `IConstructorSelector` type here. Is this something where we should technically be looking at updating the `IConstructorSelector` interface?

Reply:
If we don't want to change it, we could do that additive thing like `IConstructorSelectorSupportsSomething` and check to see if the constructor selector supports the optional feature.